### PR TITLE
Issue 2038 - Dockerfile performance improvements

### PR DESCRIPTION
**Pull request by IntelliJIN**
Reduced the number of layers by combining RUN commands.
Reduced the size of Go binaries by removing debugging information - flags-ldflags="-s -w".
Reduced the size of Go binaries by using Ultimate Packer for eXecutables (UPX). 
Added commands --no-install-recommends and rm -rf /var/lib/apt/lists/* to reduce image size.

Signed-off-by: Igor Kolinko <ikolin@softserveinc.com>

### CODE REVIEW

**Review by linggao — APPROVED** [review]

**Review by linggao — APPROVED** [review]

**Comment by linggao** [comment]
Please resolve conflicts.

**Review by linggao — DISMISSED** [review]
Please resolve the file conflicts.

**Review by linggao — CHANGES_REQUESTED** [review]
test failed. Please make sure the following works.
cd test
make realclean
make
make test

File: Makefile [hunk 1]
```diff
@@ -244,6 +250,15 @@ MAC_PKG = pkg/mac/build/horizon-cli-$(MAC_PKG_VERSION).pkg
 APT_REPO_HOST ?= 169.45.88.181
 APT_REPO_DIR ?= /vol/aptrepo-local/repositories/view-public
 
+upx:
```
Comment (dabooz) [inline Makefile:253]: Please make the use of upx optional, and by default not used.

**Review by dabooz — CHANGES_REQUESTED** [review]

**Pull request closed without merging**